src/OpenTelemetry.Instrumentation.AspNetCore/MeterProviderBuilderExtensions.cs

```diff
@@ -78,6 +78,10 @@ public static class MeterProviderBuilderExtensions
 
         builder.AddMeter(AspNetCoreMetrics.InstrumentationName);
 
+        builder.AddView(
```

Comment:
This should be done by the application user, not the instrumentation library. Instrumentation Library can set Hint (once that feature is available), but it should not try to Configure View. View is a SDK concept, and instrumentation library should not have sdk dependency at all. (its an issue that this library has an undesired dependency on sdk, thats tracked separately)...

Reply:
Experimental spec for Hint API ("instrument advice") is here:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice

Seems that Trask is ready to move it to Stable:
https://github.com/open-telemetry/opentelemetry-specification/issues/3391

I had a brief chat with Utkarsh about this. 
Seems the Hint API will be a blocker for this PR.

Reply:
Hint API require DiagnosticSource changes, so not coming this year.
But i don't know if Hint API support is required before making this change. As users always have View API to adjust bounds.

Reply:
IMO, that seems like a bad user experience.
If the Instrumentation library ships with the wrong buckets and users need to manually adjust.

Reply:
@cijothomas On thinking a bit more about this, I think it should be okay for the instrumentation library to use `View` until the Hint API is offered from DiagnosticSource. When the Hint API is available, the instrumentation library could get rid of `View` and switch to using the Hint API.

As of today, anyone using AspNetCoreInstrumentation library is going to take a dependency on the SDK anyway.

Reply:
I can include a TODO here to call out that this needs to be changed to Hint API when available.

Reply:
> I think it should be okay for the instrumentation library to use `View` until the Hint API is offered from DiagnosticSource. When the Hint API is available, the instrumentation library could get rid of `View` and switch to using the Hint API.
>
> As of today, anyone using AspNetCoreInstrumentation library is going to take a dependency on the SDK anyway.

I would open a dedicated issue to discuss and record this decision. It's a pretty fundamental principle that instrumentations should only depend on the API and not on the SDK, and so you'll want to document the rationale clearly in this case and make sure your entire community is onboard and understands the ramifications.

Reply:
> 

It can still cause other issues as Views are additive. If user adds another view matching the same metric, to drop an unwanted attribute, it'll result in 2 metrics stream. 

Its a bug that the instrumentation has a SDK dependency. It is fixable (once we move the things the instrumentation need from SDK to API itself.). Its even worse if the instrumentation leverages the SDK dependency to define behavior, reserved for application owner to define.

There are other instrumentations where this trick/hack of using views from SDK cannot be used  - like asp.net framework.

At the cost of exposing yet another public API, an option like `RecordDurationInMillisecond` could work, and is probably more desirable than misusing SDK dependency.